spec/models/market_model_spec.rb

```diff
@@ -3,6 +3,21 @@
 RSpec.describe MarketModel do
   it { expect(subject).to validate_presence_of(:name) }
 
+  it "validates market model interactions" do
```

Comment:
This could do with being more specific. Since there are multiple key/values which fail validation, you can't be sure that any one of them are responsible for the failure. I also think that `{"tariff_type" => "fixed"}` is not a valid interaction; the other attributes are required.

Consider something like:

```ruby
RSpec.describe MarketModel do
  context 'when validating' do
    let(:market)      { build(:market_model, interactions: [interaction]) }
    let(:interaction) { valid_interaction }
    let(:messages)    { market.valid? ; market.errors[:interactions] }

    MarketModel::PRESENTABLES.each do |attribute|
      context "an interaction with no '#{ attribute }'" do
        let(:interaction) { super().merge(attribute => '') }

        it 'is not valid' do
          expect(market).to_not be_valid
        end

        it 'has an error on :interactions' do
          expect(messages).to include("value for #{ attribute } can't be blank")
        end
      end

      context "an interaction with a missing '#{ attribute }'" do
        let(:interaction) { super().except(attribute) }

        it 'is not valid' do
          expect(market).to_not be_valid
        end

        it 'has an error on :interactions' do
          expect(messages).to include("value for #{ attribute } can't be blank")
        end
      end
    end # each PRESENTABLES
  end
end
```

... this is way more precise, and more useful for regression testing. For example, if one day a validation on "stakeholder_from" were to stop working, we'll catch it because the test *for that attribute* will fail (and it will be immediately obvious which attribute has a problem). The original test would continue to pass because one of the other attributes would correctly fail the validation.

It also has nice output:

```
$ rspec -fd

MarketModel
  when validating
    an interaction with no 'stakeholder_from'
      has an error on :interactions
      should not be valid
    an interaction with a missing 'stakeholder_from'
      has an error on :interactions
      should not be valid
    an interaction with no 'foundation'
      has an error on :interactions
      should not be valid
    an interaction with a missing 'foundation'
      has an error on :interactions
      should not be valid
    an interaction with no 'tariff'
      has an error on :interactions
      should not be valid
    # ...
```

It would be nice for the message to be [a bit more human-friendly](http://guides.rubyonrails.org/i18n.html#error-message-scopes), or at the very least use `key.humanize.downcase`.